packages/medusa/src/interfaces/payment-service.ts

```diff
@@ -0,0 +1,120 @@
+import { TransactionBaseService } from "./transaction-base-service"
+import {
+  Cart,
+  Customer,
+  Payment,
+  PaymentSession,
+  PaymentSessionStatus,
+} from "../models"
+import { PaymentService } from "medusa-interfaces"
+
+export type Data = Record<string, unknown>
+export type PaymentData = Data
+export type PaymentSessionData = Data
+
+export interface PaymentService<T extends TransactionBaseService<never>>
+  extends TransactionBaseService<T> {
+  getIdentifier(): string
+
+  getPaymentData(paymentSession: PaymentSession): Promise<PaymentData>
+
+  updatePaymentData(
+    paymentSessionData: PaymentSessionData,
+    data: Data
+  ): Promise<PaymentSessionData>
+
+  createPayment(cart: Cart): Promise<PaymentSessionData>
+
+  retrievePayment(paymentData: PaymentData): Promise<Data>
+
+  updatePayment(
+    paymentSessionData: PaymentSessionData,
+    cart: Cart
+  ): Promise<PaymentSessionData>
+
+  authorizePayment(
+    paymentSession: PaymentSession,
+    context: Data
+  ): Promise<{ data: PaymentSessionData; status: PaymentSessionStatus }>
+
+  capturePayment(payment: Payment): Promise<PaymentData>
+
+  refundPayment(payment: Payment, refundAmount: number): Promise<PaymentData>
+
+  cancelPayment(payment: Payment): Promise<PaymentData>
+
+  deletePayment(paymentSession: PaymentSession): Promise<void>
+
+  retrieveSavedMethods(customer: Customer): Promise<Data[]>
+
+  getStatus(
+    paymentSessionData: PaymentSessionData
```

Comment:
todo: should be `paymentData: PaymentData` to match with signature of `retrievePayment`

Reply:
This method expect a session data and not a payment data. It is always a session that is fetched and then those data are used

Reply:
I think this is why I want to have a clearer distinction between the types 😅 

We are talking about the same thing. Here we use `payment_session.data`